src/python/doc/datasets_generators.rst

```diff
@@ -0,0 +1,97 @@
+
+:orphan:
+
+.. To get rid of WARNING: document isn't included in any toctree
+
+===========================
+Datasets generators manual
+===========================
+
+We provide the generation of different customizable datasets to use as inputs for Gudhi complexes and data structures.
+
+
+Points generators
+------------------
+
+Points on sphere
+^^^^^^^^^^^^^^^^
+
+The module **_points** enables the generation of random i.i.d. points uniformly on a (d-1)-sphere in :math:`R^d`.
+The user should provide the number of points to be generated on the sphere :code:`n_samples` and the ambient dimension :code:`ambient_dim`.
+The :code:`radius` of sphere is optional and is equal to **1** by default.
+Only random points generation is currently available.
+
+The generated points are given as an array of shape :math:`(n\_samples, ambient\_dim)`.
+
+Example
+"""""""
+
+.. code-block:: python
+
+   from gudhi.datasets.generators import _points
+   from gudhi import AlphaComplex
+
+   # Generate 50 points on a sphere in R^2
+   gen_points = _points.sphere(n_samples = 50, ambient_dim = 2, radius = 1, sample = "random")
+
+   # Create an alpha complex from the generated points
+   alpha_complex = AlphaComplex(points = gen_points)
+   
+.. autofunction:: gudhi.datasets.generators._points.sphere
+
+Points on torus
+^^^^^^^^^^^^^^^^
+
+You can also generate points on a torus.
+
+Two modules are available and give the same output: the first one depends on **CGAL** and the second does not and consists of full python code.
```

Comment:
Is there a good reason to provide both? Is the one using CGAL significantly faster?
For something like wasserstein_distance, we provide 2 implementations because they provide quite different performance and approximation guarantees, and even the functionality is a bit different. But in general, it is rather confusing to have 2 versions of the same function, without explaining in the doc how the user should choose.

Reply:
I don't think that there would be a good reason to have both unless we would want to use the same code as in C++ because it already exists (and prefer having one code to maintain). From the user's point of view, I would say that the full python version is recommended. We actually need to make a choice : do we provide the C++ implementation just because it exists or we don't care and just settle for the full python one because it's enough and does the job.